### PR TITLE
removed dbt flags and updated dbt runner

### PR DESCRIPTION
**Pull request by AmandeepSinghCS**

### CODE REVIEW

File: clouddq/main.py [hunk 1]
```diff
@@ -284,26 +248,10 @@ def main(  # noqa: C901
...
-        )
-    if (not dbt_profiles_dir and (not gcp_project_id or not gcp_bq_dataset_id)) or (
-        dbt_profiles_dir and (gcp_project_id or gcp_bq_dataset_id)
-    ):
+
+    if not gcp_project_id or not gcp_bq_dataset_id:
         raise ValueError(
-            "CLI input must define connection configs in either a single dbt profiles.yml file "
-            "in '--dbt_profiles_dir' or individually using the parameters: "
+            "CLI input must define connection configs  individually using the parameters: "
```
Comment (nauh) [inline clouddq/main.py:254]:
```
CLI input must define connection configs using the parameters:
```

File: clouddq/main.py [hunk 2]
```diff
@@ -337,8 +283,18 @@ def main(  # noqa: C901
         dbt_path = dbt_runner.get_dbt_path()
         dbt_rule_binding_views_path = dbt_runner.get_rule_binding_view_path()
         dbt_entity_summary_path = dbt_runner.get_entity_summary_path()
-        dbt_profiles_dir = dbt_runner.get_dbt_profiles_dir()
-        environment_target = dbt_runner.get_dbt_environment_target()
+        dbt_profiles_dir = dbt_runner.get_dbt_profiles_dir(
```
Comment (nauh) [inline clouddq/main.py:286]: If I am not mistaken, `_resolve_connection_configs` is effectively called twice (once in `get_dbt_profiles_dir` and once in `get_dbt_environment_target`). Why do we need to do this? Or it means we should refactor the code?

File: clouddq/runners/dbt/dbt_runner.py [hunk 3]
```diff
@@ -123,93 +117,93 @@ def get_entity_summary_path(self) -> Path:
...
-            )
-        else:
-            dbt_path = Path(dbt_path).absolute()
-            if dbt_path.name != "dbt":
-                dbt_path = dbt_path / "dbt"
+        dbt_path = Path().cwd().joinpath("dbt").absolute()
+        logger.debug(
+            "No argument 'dbt_path' provided. Defaulting to use "
+            f"'dbt' directory in current working directory at: {dbt_path}"
+        )
```
Comment (nauh) [inline clouddq/runners/dbt/dbt_runner.py:206]: Should we update the logging message?

File: clouddq/main.py (hunk 1 above)
Comment (AmandeepSinghCS) [reply clouddq/main.py:254]: incorporated

File: clouddq/runners/dbt/dbt_runner.py (hunk 3 above)
Comment (AmandeepSinghCS) [reply clouddq/runners/dbt/dbt_runner.py:206]: updated to: **Defaulting to use 'dbt' directory in current working directory at: {dbt_path}**

File: clouddq/main.py (hunk 2 above)
Comment (AmandeepSinghCS) [reply clouddq/main.py:286]: I think, we should refactor the code a bit.

File: clouddq/main.py [hunk 4]
```diff
@@ -337,8 +283,21 @@ def main(  # noqa: C901
...
+            gcp_region_id=gcp_region_id,
+            bigquery_client=bigquery_client,
+        ).get("dbt_profiles_dir")
+
+        environment_target = dbt_runner.get_dbt_profiles_dir_and_environment_target(
+            gcp_project_id=gcp_project_id,
+            gcp_bq_dataset_id=gcp_bq_dataset_id,
+            gcp_region_id=gcp_region_id,
+            bigquery_client=bigquery_client,
+        ).get("environment_target")
```
Comment (thinhha) [inline clouddq/main.py:299]: suggest to return a tuple from get_dbt_profiles_dir_and_environment_target() and calling it once like this:

```
dbt_profiles_dir, environment_target = dbt_runner. get_dbt_profiles_dir_and_environment_target(...)
```

File: clouddq/runners/dbt/dbt_runner.py [hunk 5]
```diff
@@ -123,93 +117,79 @@ def get_entity_summary_path(self) -> Path:
...
-                gcp_project_id=gcp_project_id,
-                gcp_bq_dataset_id=gcp_bq_dataset_id,
-                bigquery_client=bigquery_client,
-                gcp_region_id=gcp_region_id,
-                gcp_service_account_key_path=gcp_service_account_key_path,
-                gcp_impersonation_credentials=gcp_impersonation_credentials,
+            self.connection_config.to_dbt_profiles_yml(
+                target_directory=self.dbt_profiles_dir,
+                environment_target=self.environment_target,
             )
```
Comment (thinhha) [inline clouddq/runners/dbt/dbt_runner.py:178]: this can be moved outside of the if else and called only once

File: scripts/test_pip_install.sh [hunk 6]
```diff
@@ -110,11 +110,16 @@ sed -i s/\<my-gcp-project-id\>/"${GOOGLE_CLOUD_PROJECT}"/g "$TEST_DIR"/configs/m
...
     --debug  \
     --dry_run
-python3 -m clouddq T1_DQ_1_VALUE_NOT_NULL,T2_DQ_1_EMAIL,T3_DQ_1_EMAIL_DUPLICATE configs --dbt_profiles_dir="$TEST_DIR" \
-    --dbt_path="$TEST_DIR"  \
+python3 -m clouddq T1_DQ_1_VALUE_NOT_NULL,T2_DQ_1_EMAIL,T3_DQ_1_EMAIL_DUPLICATE configs \
+    --gcp_project_id="${GOOGLE_CLOUD_PROJECT}" \
+    --gcp_bq_dataset_id="${CLOUDDQ_BIGQUERY_DATASET}" \
+    --gcp_region_id="${CLOUDDQ_BIGQUERY_REGION}" \
     --debug  \
     --dry_run
```
Comment (thinhha) [inline scripts/test_pip_install.sh:124]: we can remove all of this

File: tests/integration/test_cli_integration.py [hunk 7]
```diff
@@ -66,25 +66,29 @@ def test_cli_dbt_path(
...
-                    f"--dbt_profiles_dir={test_profiles_dir}",
+                    f"--gcp_project_id={gcp_project_id}",
+                    f"--gcp_bq_dataset_id={gcp_bq_dataset}",
+                    f"--gcp_region_id={gcp_bq_region}",
                     "--dry_run",
-                    "--debug",
+                    "--debug"
                     ]
                 result = runner.invoke(main, args)
                 print(result.output)
```
Comment (thinhha) [inline tests/integration/test_cli_integration.py:94]: aren't we already running this test in test_cli_unit?

File: tests/unit/test_cli_unit.py [hunk 8]
```diff
@@ -146,7 +154,7 @@ def test_cli_dry_run_incompatiable_arguments_failure(
                 result = runner.invoke(main, args)
                 print(result.output)
                 assert result.exit_code == 1
-                assert isinstance(result.exception, ValueError)
+                assert isinstance(result.exception, SystemExit)
         finally:
             shutil.rmtree(temp_dir)
```
Comment (thinhha) [inline tests/unit/test_cli_unit.py:159]: do we still need this test?

File: clouddq/main.py (hunk 4 above)
Comment (AmandeepSinghCS) [reply clouddq/main.py]: incorporated

File: clouddq/runners/dbt/dbt_runner.py (hunk 5 above)
Comment (AmandeepSinghCS) [reply clouddq/runners/dbt/dbt_runner.py]: updated.

File: scripts/test_pip_install.sh (hunk 6 above)
Comment (AmandeepSinghCS) [reply scripts/test_pip_install.sh]: removed.

**Review by thinhha — APPROVED** [review]
LGTM - thanks!

File: clouddq/runners/dbt/dbt_runner.py [hunk 9]
```diff
@@ -123,93 +117,77 @@ def get_entity_summary_path(self) -> Path:
...
-            dbt_profiles_dir=self.dbt_profiles_dir,
-            environment_target=self.environment_target,
+        return (
+            Path(self.dbt_profiles_dir),
+            self.environment_target
         )
-        return self.environment_target
 
-    def get_dq_summary_dataset_region(self) -> str:
+    def get_dq_summary_dataset_region(self, gcp_project_id, gcp_bq_dataset_id) -> str:
```
[the remark this answers is not in the document]
Comment (AmandeepSinghCS) [reply clouddq/runners/dbt/dbt_runner.py:139]: I checked, it's not getting used now.

**Review by nauh — APPROVED** [review]

**Pull request merged**